File: Sprint6/B/C++/main.cpp
// https://contest.yandex.ru/contest/25069/problems/B/
#include <iostream>
#include <vector>

int main() {
    int n, m;
    std::cin >> n >> m;

    std::vector<std::vector<int>> matrix(n, std::vector<int>(n, 0));

    for (int i = 0; i < m; ++i) {
        int u, v;
        std::cin >> u >> v;
        matrix[u - 1][v - 1] = 1;
    }

    for (const auto& row : matrix) {
        for (int val : row) {
            std::cout << val << " ";
        }
        std::cout << std::endl;
    }

    return 0;
}

File: Sprint6/B/Rust/src/main.rs
// https://contest.yandex.ru/contest/25069/problems/B/

use std::io::{self, BufRead};

fn main() -> io::Result<()> {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();

    let first_line = lines.next().unwrap().unwrap();
    let nm: Vec<usize> = first_line
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();
    let (n, m) = (nm[0], nm[1]);

    let mut matrix = vec![vec![0; n]; n];

    for _ in 0..m {
        let edge: Vec<usize> = lines
            .next()
            .unwrap()
            .unwrap()
            .split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();
        let (u, v) = (edge[0], edge[1]);

        matrix[u - 1][v - 1] = 1;
    }

    for row in &matrix {
        for &val in row {
            print!("{} ", val);
        }
        println!();
    }

    Ok(())
}
